### PR TITLE
tentative solution for multiple calls of `analyze_vars_in_cols`

### LINKED ISSUE #931: [Question]: summarize_vars_in_cols no longer exists
### What is your question?

Hi Nest folks, 

In older versions (< 0.8.0) of tern I was using `summarize_vars_in_cols` but I just saw in the later versions this function is no longer an exported object (breaking change for me). Is there any alternative so that we can summarize the statistics in columns? 

This was the [file](https://github.com/insightsengineering/tern/blob/v0.7.10/R/summarize_variables_in_cols.R)
Thanks, 

IZ

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #931 and #936

### CODE REVIEW

File: R/analyze_vars_in_cols.R [hunk 1]
```diff
@@ -136,15 +177,53 @@ analyze_vars_in_cols <- function(lyt,
...
+      nested = nested,
+      extra_args = list(...)
+    )
+  }
+}
+
+# TODO: multiple analyze calls WITHOUT the row split
+
+# Help function
+get_last_col_split <- function(lyt) {
```
Comment (Melkiades) [inline R/analyze_vars_in_cols.R:227]: This help function will go eventually in rtables @gmbecker

File: tests/testthat/test-survival_timepoint.R [hunk 2]
```diff
@@ -235,4 +235,68 @@ testthat::test_that("surv_timepoint no warning when multipled layers generated",
...
+      .stats = "count_fraction",
+      .labels = c(count_fraction = "Patients with event (%)")
+    ) %>%
+    summarize_vars(
+      vars = "is_not_event",
+      .stats = "count_fraction",
+      .labels = c(count_fraction = "Patients without event (%)"),
+      nested = FALSE,
+      show_labels = "hidden"
+      # The following creates an error: .indent_mods = 1L (TODO: check this)
```
Comment (Melkiades) [inline tests/testthat/test-survival_timepoint.R:265]: This needs to be addressed by https://github.com/insightsengineering/tern/issues/941 @edelarua

File: tests/testthat/test-survival_timepoint.R [hunk 3]
```diff
@@ -235,4 +235,68 @@ testthat::test_that("surv_timepoint no warning when multipled layers generated",
...
+    ) %>%
+    surv_timepoint(
+      vars = "AVAL",
+      var_labels = "Months",
+      time_point = c(6, 12),
+      is_event = "is_event",
+      method = "both",
+      table_names_suffix = "3",
+      control = control_surv_timepoint(),
+      .indent_mods = 1L # These do nothing atm (TODO: fix this behavior)
```
Comment (Melkiades) [inline tests/testthat/test-survival_timepoint.R:295]: All of these do not work. Issue for https://github.com/insightsengineering/tern/issues/941 @edelarua

**Comment by Melkiades** [comment]
I tag you @edelarua @ayogasekaram @shajoezhu to take a look at the current solution (see tests). I think the output is finally what is needed for PKCT01. There is a problem with `labelstr` for `summarize_row_groups` (it does not do it), but I am working on it by implementing another solution for this on the side that was suggested by Gabe. I will file issues to `rtables` if this still does not work

File: R/analyze_vars_in_cols.R [hunk 4]
```diff
@@ -136,15 +151,129 @@ analyze_vars_in_cols <- function(lyt,
...
+    analyze_colvars(lyt,
+      afun = afun_list,
+      nested = nested,
+      extra_args = list(...)
+    )
+  }
+}
+
+# TODO: multiple analyze calls WITHOUT the row split
+
```
Comment (Melkiades) [inline R/analyze_vars_in_cols.R:275]: still to try but preliminary tests do not seem to work and only the first analyze is used

File: R/survival_timepoint.R [hunk 5]
```diff
@@ -339,6 +339,7 @@ surv_timepoint <- function(lyt,
         var_labels = paste(tpt, var_labels),
         table_names = paste0("surv_diff_", tpt, table_names_suffix),
         show_labels = ifelse(method == "both", "hidden", show_labels),
+        indent_mod = ifelse(method == "both", 1L, 0L),
```
[the remark this answers is not in the document]
Comment (edelarua) [reply R/survival_timepoint.R:342]: @Melkiades this bug is addressed in this commit: https://github.com/insightsengineering/tern/pull/945/commits/13c92ba1ab8a3346be1b126ace0564b9390f242d

There are several functions in `tern` which accept a vector for `.indent_mods` with each element corresponding to one statistic row's indentation (presumably there are situations where different statistics should have different indentation levels), so indentation levels need to be specified by stat in this function. Some other functions with this behaviour are `compare_vars` and `summarize_vars`.

File: tests/testthat/test-survival_timepoint.R (hunk 3 above)
Comment (edelarua) [reply tests/testthat/test-survival_timepoint.R:295]: See my above comment again.

**Comment by Melkiades** [comment]
This is ready. I will create corresponding fixes in `scda.test` and catalog. Please consider comparing this with `summarize_patients_exposure_in_cols` and `analyze_patients_exposure_in_cols` which do a similar thing but with much more nesting of functions and duplication

**Review by edelarua — APPROVED** [review]
Lgtm! I tried it out and it seems to work fine. Thanks Davide!

**Pull request merged**